### ISSUE TITLE
Image Size limitation for registration and cv2 error

### ISSUE THREAD

**Issue by B-Betz** [labsyspharm/palom#13]
Hello @Yu-AnChen!

I produced some amazing registrations with PALOM! 
Unfortunately, I have encountered and error for big images which I think is due to the size of these images. 
Is there I size limitation and if yes, what is the maximum size PALOM can process?
The image size for which I encounter the error is width 54977 pixel, height 45935 (physical pixel size 0.325 um)
The error:
"/path/to/palom/block_affine.py", line 102 in lock affine
 else:raise(err)
File
"/path/to/palom/palom/block_affine.py", line 90, in block_affine
  warped_src_img_block = cv.2warAffine(
 cv2.error:OpenCV(4.8.0)/io/opencv/modules/imgproc/src/imgwarp.cpp:1730:error: (-215:Assertion failed) dst.cols <SHRT_MAX && dst.rows <SHRT_MAX && src.cols <SHRT_MAX && src.rows <SHRT_MAX in function 'remap'

The error points towards a function of the OpenCV library. If this question should be moved there let me know.
Thank you for the help!
Best Barbara

**Comment by Yu-AnChen** [comment]
Hey Babara, thanks for trying it out and reporting back. We've seen this before - did you install palom fairly recently and are you on a Linux/Unix machine or a Windows machine?

**Comment by B-Betz** [comment]
Thank you for the answer! 
I am working on a Linux computer cluster. I submit the script with enough memory, at least to my understanding. 
Here is the resource summary of the failed job:

    CPU time :                                   38.02 sec.
    Max Memory :                                 10146 MB
    Average Memory :                             2817.00 MB
    Total Requested Memory :                     61440.00 MB
    Delta Memory :                               51294.00 MB
    Max Swap :                                   -
    Max Processes :                              3
    Max Threads :                                6
    Run time :                                   276 sec.
    Turnaround time :                            278 sec.

I updated palom recently, the installed version in my environment is 2023.8.1.
Thank you very much for your help!

**Comment by Yu-AnChen** [comment]
Thanks for the details! A couple more questions 😅

- Are you running palom in a conda/mamba environment? If so could you export the env package versions ([info here](https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html#building-identical-conda-environments)) so that I could try to reproduce?

- How big is the file size? Is that something you could share for me to inspect?

- If you are comfortable with docker, palom is available in [this docker image](https://hub.docker.com/layers/labsyspharm/s3segmenter/1.5.5-large/images/sha256-192f2affb6401646a6c7c17d801f4f7b14e819a409690dba2b1dc9e4c2a42ac9?context=explore). You could use it to test if the error persists.

**Comment by B-Betz** [comment]
Hi!

Yes I am running it in a conda environment,here is the export: 
[spec-file_palom_env.txt](https://github.com/labsyspharm/palom/files/12573134/spec-file_palom_env.txt)
The sizes of the images are Cycle 1 34 GB, Cycle 2 20 GB, Cycle 3 28 GB. I successfully registered Cycle 1 and Cycle 2, but the error occurs when I register Cycle 3 to 1 and 2. 
I unfortunately cannot share them at the moment, this might change. I understand if this makes it more difficult to find the error. 
It is not possible for me to use docker directly on our computer cluster but singularity. I will try to pull it through singularity!
Thank you for the suggestion!

**Comment by B-Betz** [comment]
The script has run through without any errors and the registration looks good to me!
I think the way the script is processing the images (3 at a time) makes a different? When I got the error initially, I have registered cycle 1 and 2, and afterwards, I performed registration of cycle 3 to the already registered image consisting of cycle 1 and 2. 
Attached you find the QC plots, it looks to me like the second image is rotated on the plot (which shouldn't be the case based on the raw image), but the registration worked and looks good!
I used the script to register the whole experiment (5 Cycles) and the registration looks good for all Cycles.
So this has solved my problem, thank you very much for your help!

![Cycle2_ome tiff (block shift distance)](https://github.com/labsyspharm/palom/assets/133800785/23dc9d7e-2850-4b62-aaec-3a52b9e0aec7)
![Cycle2_ome tiff (coarse alignment)](https://github.com/labsyspharm/palom/assets/133800785/a5f40beb-f1bf-4bda-9097-aa7bed9de7ca)

![Cycle3_ome tiff (block shift distance)](https://github.com/labsyspharm/palom/assets/133800785/ab29cc65-e79a-4560-aa6e-27a8788583cc)
![Cycle3_ome tiff (coarse alignment)](https://github.com/labsyspharm/palom/assets/133800785/1e0e60e2-87fe-4bc8-acf2-a1f97013139a)
